addons/mail/static/src/models/link_preview.js

```diff
@@ -0,0 +1,95 @@
+/** @odoo-module **/
+
+import { registerModel } from '@mail/model/model_core';
+import { attr, many, one } from '@mail/model/model_field';
+import { clear } from '@mail/model/model_field_command';
+
+registerModel({
+    name: 'LinkPreview',
+    identifyingFields: ['id'],
+    recordMethods: {
+        async remove() {
+            await this.messaging.rpc({
+                route: '/mail/link_preview/delete',
+                params: { link_preview_id: this.id },
+            }, { shadow: true });
+        },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsDeletable() {
+            if (!this.messaging) {
+                return false;
+            }
+            return this.message.canBeDeleted ||
```

Comment:
We probably want to be less strict than message, especially if the preview automatically appears after, it should always be removable by the user who posted it.